tests/release-notes-test.ts

```diff
@@ -0,0 +1,68 @@
+jest.mock("danger", () => jest.fn());
+import danger from "danger";
+const dm = danger as any;
+
+import checkReleaseNotes from "../org/pr/release-notes";
+
+// The mocked data and return values for calls the rule makes.
+beforeEach(() => {
+    dm.warn = jest.fn().mockReturnValue(true);
+
+    dm.danger = {
+        git: {
+            modified_files: ["RELEASE-NOTES.txt"],
+            created_files: [],
+            deleted_files: []
+        },
+        github: {
+            pr: {
+                base: {
+                    ref: "release/1.0",
+                }
+            },
+            issue: {
+                labels: []
+            },
+        },
+    };
+});
+
+describe("release notes checks", () => {
+    it("warns when RELEASE-NOTES.txt is changed in a PR which targets a release branch", async () => {
+        await checkReleaseNotes();
+        
+        expect(dm.warn).toHaveBeenCalledWith(expect.stringContaining("This PR contains changes to RELEASE_NOTES.txt.\n"));
+    })
+
+    it("does not warn when RELEASE-NOTES.txt is changed in a PR which targets a release branch if the PR is labeled with Releases", async () => {
+        dm.danger.github.issue.labels = [
+            {
+                name: 'Releases'
+            }
+        ]
+
+        await checkReleaseNotes();
+        
+        expect(dm.warn).not.toHaveBeenCalled();
+    })
+
+    it("does not warn when RELEASE-NOTES.txt is changed in a PR which doesn't target a release branch", async () => {
+        dm.danger.github.pr.base.ref = "develop";
+
+        await checkReleaseNotes();
+        
+        expect(dm.warn).not.toHaveBeenCalled();
+    })
+
+    it("does not warn when RELEASE-NOTES.txt is not changed", async () => {
+        dm.danger.git.modified_files = [
+                "file1.swift",
+                "file2.php",
+                "file3.kt"
+        ]
+
+        await checkReleaseNotes();
+        
+        expect(dm.warn).not.toHaveBeenCalled();
+    })
+})
```

Comment:
nit: CRLF at EOF (to avoid potential oddities with git and diff and other tooling)
```suggestion
})

```

Reply:
I once did some research into the rationale for this and came up with the following pre-def comment that I sometimes add to code reviews:

The convention of having a new line at the end of the files comes from an old C decision that has been passed down through Unix history:

> A source file that is not empty shall end in a new-line character, which shall not be immediately preceded by a backslash character.
> 
> Since this is a “shall” clause, we must emit a diagnostic message for a violation of this rule.

So, it turns out that, according to POSIX, every text file (including Ruby and JavaScript source files) should end with a `\n`, or “newline” (not “a new line”) character. This acts as the `eol`, or the “end of line” character. It is a line “terminator”. ([Source](https://thoughtbot.com/blog/no-newline-at-end-of-file))

If anything, addressing these also spares us from that annoying red indicator when reading the file on GitHub. I keep talking about `.editorconfig` and how it can help with this, but never do it 😞